### PR TITLE
[WIP] New package: fluent-bit-1.7.1

### PR DESCRIPTION
**Pull request by SimonAdameit**
#### General
- [ ] This is a new package and it conforms to the [quality requirements](https://github.com/void-linux/void-packages/blob/master/Manual.md#quality-requirements)

#### Have the results of the proposed changes been tested?
- [ ] I use the packages affected by the proposed changes on a regular basis and confirm this PR works for me
- [ ] I generally don't use the affected packages but briefly tested this PR

### CODE REVIEW

File: srcpkgs/fluent-bit/template [hunk 1]
```diff
@@ -0,0 +1,38 @@
+# Template file for 'fluent-bit'
+pkgname=fluent-bit
+version=1.7.1
+revision=1
+short_desc="Fluent Bit is an open source Log Processor and Forwarder"
+maintainer="Simon Adameit <simon@simonadameit.com>"
+license="Apache-2.0"
+homepage="https://fluentbit.io/"
+distfiles="https://fluentbit.io/releases/1.7/fluent-bit-${version}.tar.gz"
```
Comment (sgn) [inline srcpkgs/fluent-bit/template:9]: `s/1.7/${version%.*}/`

File: srcpkgs/fluent-bit/template [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+	configure_args+=" -DCMAKE_C_STANDARD_LIBRARIES=-lfts"
+fi
+
+# Set release flags
+configure_args+=" -DFLB_RELEASE=On"
+configure_args+=" -DFLB_DEBUG=Off"
+configure_args+=" -DFLB_TRACE=Off"
+
+# Enable Features
+configure_args+=" -DFLB_JEMALLOC=On"
```
Comment (sgn) [inline srcpkgs/fluent-bit/template:32]: Unbundle `jemalloc` please.

Comment (sgn) [reply srcpkgs/fluent-bit/template:32]: And `mbedtls`, too.

Comment (sgn) [reply srcpkgs/fluent-bit/template:32]: In other word, I would `rm -rf lib` in `post_extract` and fix their build system to use system libraries.

File: srcpkgs/fluent-bit/patches/flb_coro.patch [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+# https://docs.fluentbit.io/manual/administration/configuring-fluent-bit/configuration-file#config_section
+#
+--- include/fluent-bit/flb_coro.h
++++ include/fluent-bit/flb_coro.h
+@@ -65,7 +65,7 @@
+ #ifdef FLB_CORO_STACK_SIZE
+ #define FLB_CORO_STACK_SIZE      FLB_CORO_STACK_SIZE
+ #else
+-#define FLB_CORO_STACK_SIZE      ((3 * PTHREAD_STACK_MIN) / 2)
++#define FLB_CORO_STACK_SIZE      24576
```
Comment (sgn) [inline srcpkgs/fluent-bit/patches/flb_coro.patch:15]: No need to patch, either below options should be fine.
```sh
configure_args+="-DFLB_CORO_STACK_SIZE=24576"
```
or:
```sh
CFLAGS=-DFLB_CORO_STACK_SIZE=24576
```

**Comment by SimonAdameit** [comment]
@sgn @ericonr Thanks for the reviews! I decided to not get this into void, due to the effort involved. I still applied and pushed the changes you suggested, except for the unbundling, which I won't perform now.

**Comment by sgn** [comment]
Hi, I spent my time to de-bundle most of libraries from fluent-bit.
If you want to try, here you're https://github.com/sgn/void-packages/tree/fluent-bit

If you can confirm it's good for you, I'll submit it to upstream.
There're some libraries that I didn't de-bundle because their developers also work on fluent-bit.

**Pull request closed without merging**